packages/docusaurus/src/webpack/plugins/CleanWebpackPlugin.ts

```diff
@@ -152,6 +166,12 @@ export default class CleanWebpackPlugin {
       return;
     }
 
+    if (this.checkIfFileNameIsDuplicate()) {
+      throw new Error(
+        `Output directory ${this.outputPath} already exists. Docusaurus needs this directory to save the build output. Either remove the directory or choose a different build directory via '--out-dir'.`,
```

Comment:
If it's a directory the role of this plugin is to delete. As @Josh-Cena said the problem is only if a file already exists, so this error message is not good.

Reply:
@slorber this makes sense, I'll do the change

Reply:
changes done